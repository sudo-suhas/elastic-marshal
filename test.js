'use strict';

/* eslint-env jest */

const bob = require('elastic-builder');
const cases = require('jest-in-case');
const marshaller = require('./');

cases(
    '[un]marshal',
    opts => {
        const qryStr = marshaller.marshal(opts.qry);
        expect(typeof qryStr).toBe('string');
        expect(qryStr).toMatchSnapshot();
        const unmarshalled = marshaller.unmarshal(qryStr);
        expect(unmarshalled).toBeInstanceOf(opts.constructor);
        expect(unmarshalled).toEqual(opts.qry);
    },
    [
        {
            name: 'BoolQuery',
            qry: new bob.BoolQuery()
                .must(new bob.TermQuery('user', 'kimchy'))
                .must(new bob.RangeQuery('age').gte(10).lte(20))
                .must(
                    new bob.BoolQuery().should([
                        new bob.TermQuery('tag', 'wow'),
                        new bob.TermQuery('tag', 'elasticsearch')
                    ])
                ),
            constructor: bob.BoolQuery
        },
        {
            name: 'RequestBodySearch with query',
            qry: new bob.RequestBodySearch()
                .query(new bob.MatchQuery('message', 'this is a test'))
                .from(0)
                .size(10),
            constructor: bob.RequestBodySearch
        },
        {
            name: 'RequestBodySearch with query and aggregation',
            qry: new bob.RequestBodySearch()
                .query(new bob.MatchQuery('message', 'this is a test'))
                .agg(
                    new bob.TermsAggregation('towns', 'town').agg(
                        new bob.GeoCentroidAggregation('centroid', 'location')
                    )
                )
                .from(0)
                .size(10),
            constructor: bob.RequestBodySearch,
            skip: true
        },
        {
            name: 'DateHistogramAggregation',
            qry: new bob.DateHistogramAggregation('sales_per_month', 'date')
                .interval('month')
                .agg(new bob.SumAggregation('sales', 'price'))
                .agg(new bob.DerivativeAggregation('sales_deriv', 'sales'))
                .agg(
                    new bob.DerivativeAggregation(
                        'sales_2nd_deriv',
                        'sales_deriv'
                    )
                ),
            constructor: bob.DateHistogramAggregation,
            skip: true
        }
    ]
);
