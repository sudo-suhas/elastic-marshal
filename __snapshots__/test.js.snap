// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[un]marshal BoolQuery 1`] = `"[[{\\"c\\":\\"BoolQuery\\",\\"d\\":{\\"queryType\\":\\"bool\\",\\"_body\\":{\\"o\\":1},\\"_queryOpts\\":{\\"o\\":2}}},{\\"c\\":\\"Object\\",\\"d\\":{\\"bool\\":{\\"o\\":2}}},{\\"c\\":\\"Object\\",\\"d\\":{\\"must\\":{\\"a\\":[{\\"o\\":3},{\\"o\\":6},{\\"o\\":9}]}}},{\\"c\\":\\"TermQuery\\",\\"d\\":{\\"queryType\\":\\"term\\",\\"_body\\":{\\"o\\":4},\\"_queryOpts\\":{\\"o\\":5},\\"_field\\":\\"user\\"}},{\\"c\\":\\"Object\\",\\"d\\":{\\"term\\":{\\"o\\":5}}},{\\"c\\":\\"Object\\",\\"d\\":{\\"value\\":\\"kimchy\\"}},{\\"c\\":\\"RangeQuery\\",\\"d\\":{\\"queryType\\":\\"range\\",\\"_body\\":{\\"o\\":7},\\"_queryOpts\\":{\\"o\\":8},\\"_field\\":\\"age\\"}},{\\"c\\":\\"Object\\",\\"d\\":{\\"range\\":{\\"o\\":8}}},{\\"c\\":\\"Object\\",\\"d\\":{\\"gte\\":10,\\"lte\\":20}},{\\"c\\":\\"BoolQuery\\",\\"d\\":{\\"queryType\\":\\"bool\\",\\"_body\\":{\\"o\\":10},\\"_queryOpts\\":{\\"o\\":11}}},{\\"c\\":\\"Object\\",\\"d\\":{\\"bool\\":{\\"o\\":11}}},{\\"c\\":\\"Object\\",\\"d\\":{\\"should\\":{\\"a\\":[{\\"o\\":12},{\\"o\\":15}]}}},{\\"c\\":\\"TermQuery\\",\\"d\\":{\\"queryType\\":\\"term\\",\\"_body\\":{\\"o\\":13},\\"_queryOpts\\":{\\"o\\":14},\\"_field\\":\\"tag\\"}},{\\"c\\":\\"Object\\",\\"d\\":{\\"term\\":{\\"o\\":14}}},{\\"c\\":\\"Object\\",\\"d\\":{\\"value\\":\\"wow\\"}},{\\"c\\":\\"TermQuery\\",\\"d\\":{\\"queryType\\":\\"term\\",\\"_body\\":{\\"o\\":16},\\"_queryOpts\\":{\\"o\\":17},\\"_field\\":\\"tag\\"}},{\\"c\\":\\"Object\\",\\"d\\":{\\"term\\":{\\"o\\":17}}},{\\"c\\":\\"Object\\",\\"d\\":{\\"value\\":\\"elasticsearch\\"}}],{\\"o\\":0}]"`;

exports[`[un]marshal DateHistogramAggregation 1`] = `"[[{\\"c\\":\\"DateHistogramAggregation\\",\\"d\\":{\\"_name\\":\\"sales_per_month\\",\\"aggType\\":\\"date_histogram\\",\\"_aggs\\":{\\"o\\":1},\\"_aggsDef\\":{\\"o\\":2},\\"_nestedAggs\\":{\\"a\\":[{\\"o\\":3},{\\"o\\":6},{\\"o\\":9}]}}},{\\"c\\":\\"Object\\",\\"d\\":{\\"date_histogram\\":{\\"o\\":2}}},{\\"c\\":\\"Object\\",\\"d\\":{\\"field\\":\\"date\\",\\"interval\\":\\"month\\"}},{\\"c\\":\\"SumAggregation\\",\\"d\\":{\\"_name\\":\\"sales\\",\\"aggType\\":\\"sum\\",\\"_aggs\\":{\\"o\\":4},\\"_aggsDef\\":{\\"o\\":5},\\"_nestedAggs\\":{\\"a\\":[]}}},{\\"c\\":\\"Object\\",\\"d\\":{\\"sum\\":{\\"o\\":5}}},{\\"c\\":\\"Object\\",\\"d\\":{\\"field\\":\\"price\\"}},{\\"c\\":\\"DerivativeAggregation\\",\\"d\\":{\\"_name\\":\\"sales_deriv\\",\\"aggType\\":\\"derivative\\",\\"_aggs\\":{\\"o\\":7},\\"_aggsDef\\":{\\"o\\":8},\\"_nestedAggs\\":{\\"a\\":[]},\\"_refUrl\\":\\"https://www.elastic.co/guide/en/elasticsearch/reference/current/search-aggregations-pipeline-derivative-aggregation.html\\"}},{\\"c\\":\\"Object\\",\\"d\\":{\\"derivative\\":{\\"o\\":8}}},{\\"c\\":\\"Object\\",\\"d\\":{\\"buckets_path\\":\\"sales\\"}},{\\"c\\":\\"DerivativeAggregation\\",\\"d\\":{\\"_name\\":\\"sales_2nd_deriv\\",\\"aggType\\":\\"derivative\\",\\"_aggs\\":{\\"o\\":10},\\"_aggsDef\\":{\\"o\\":11},\\"_nestedAggs\\":{\\"a\\":[]},\\"_refUrl\\":\\"https://www.elastic.co/guide/en/elasticsearch/reference/current/search-aggregations-pipeline-derivative-aggregation.html\\"}},{\\"c\\":\\"Object\\",\\"d\\":{\\"derivative\\":{\\"o\\":11}}},{\\"c\\":\\"Object\\",\\"d\\":{\\"buckets_path\\":\\"sales_deriv\\"}}],{\\"o\\":0}]"`;

exports[`[un]marshal RequestBodySearch with query 1`] = `"[[{\\"c\\":\\"RequestBodySearch\\",\\"d\\":{\\"_body\\":{\\"o\\":1},\\"_aggs\\":{\\"a\\":[]},\\"_suggests\\":{\\"a\\":[]}}},{\\"c\\":\\"Object\\",\\"d\\":{\\"query\\":{\\"o\\":2},\\"from\\":0,\\"size\\":10}},{\\"c\\":\\"MatchQuery\\",\\"d\\":{\\"queryType\\":\\"match\\",\\"_body\\":{\\"o\\":3},\\"_queryOpts\\":{\\"o\\":4},\\"_field\\":\\"message\\"}},{\\"c\\":\\"Object\\",\\"d\\":{\\"match\\":{\\"o\\":4}}},{\\"c\\":\\"Object\\",\\"d\\":{\\"query\\":\\"this is a test\\"}}],{\\"o\\":0}]"`;

exports[`[un]marshal RequestBodySearch with query and aggregation 1`] = `"[[{\\"c\\":\\"RequestBodySearch\\",\\"d\\":{\\"_body\\":{\\"o\\":1},\\"_aggs\\":{\\"a\\":[{\\"o\\":5}]},\\"_suggests\\":{\\"a\\":[]}}},{\\"c\\":\\"Object\\",\\"d\\":{\\"query\\":{\\"o\\":2},\\"from\\":0,\\"size\\":10}},{\\"c\\":\\"MatchQuery\\",\\"d\\":{\\"queryType\\":\\"match\\",\\"_body\\":{\\"o\\":3},\\"_queryOpts\\":{\\"o\\":4},\\"_field\\":\\"message\\"}},{\\"c\\":\\"Object\\",\\"d\\":{\\"match\\":{\\"o\\":4}}},{\\"c\\":\\"Object\\",\\"d\\":{\\"query\\":\\"this is a test\\"}},{\\"c\\":\\"TermsAggregation\\",\\"d\\":{\\"_name\\":\\"towns\\",\\"aggType\\":\\"terms\\",\\"_aggs\\":{\\"o\\":6},\\"_aggsDef\\":{\\"o\\":7},\\"_nestedAggs\\":{\\"a\\":[{\\"o\\":8}]},\\"_refUrl\\":\\"https://www.elastic.co/guide/en/elasticsearch/reference/current/search-aggregations-bucket-terms-aggregation.html\\"}},{\\"c\\":\\"Object\\",\\"d\\":{\\"terms\\":{\\"o\\":7}}},{\\"c\\":\\"Object\\",\\"d\\":{\\"field\\":\\"town\\"}},{\\"c\\":\\"GeoCentroidAggregation\\",\\"d\\":{\\"_name\\":\\"centroid\\",\\"aggType\\":\\"geo_centroid\\",\\"_aggs\\":{\\"o\\":9},\\"_aggsDef\\":{\\"o\\":10},\\"_nestedAggs\\":{\\"a\\":[]}}},{\\"c\\":\\"Object\\",\\"d\\":{\\"geo_centroid\\":{\\"o\\":10}}},{\\"c\\":\\"Object\\",\\"d\\":{\\"field\\":\\"location\\"}}],{\\"o\\":0}]"`;
